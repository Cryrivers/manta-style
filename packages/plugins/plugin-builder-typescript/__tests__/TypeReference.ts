import getTranspiledString from '../src/utils/transpile';

describe('Type Reference', () => {
  test('Reference Type (Non-generic)', () => {
    expect(
      getTranspiledString('type A = string; type B = A;'),
    ).toMatchSnapshot();
  });
  test('Reference Type (Generic)', () => {
    expect(
      getTranspiledString('type A<T> = string | T; type B = A<number>;'),
    ).toMatchSnapshot();
  });
});
