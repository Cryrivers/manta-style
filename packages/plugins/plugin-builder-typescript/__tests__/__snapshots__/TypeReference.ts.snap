// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type Reference Reference Type (Generic) 1`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var A = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"A\\", function (typeFactory) {
    var T = typeFactory.TypeParameter(\\"T\\");
    var type = runtime_1[\\"default\\"].UnionType([runtime_1[\\"default\\"].StringKeyword, T]);
    return type;
}, []);
var B = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"B\\", function (typeFactory) {
    var type = A.argumentTypes([runtime_1[\\"default\\"].NumberKeyword]);
    return type;
}, []);
"
`;

exports[`Type Reference Reference Type (Non-generic) 1`] = `
"\\"use strict\\";
exports.__esModule = true;
var runtime_1 = require(\\"@manta-style/runtime\\");
var A = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"A\\", function (typeFactory) {
    var type = runtime_1[\\"default\\"].StringKeyword;
    return type;
}, []);
var B = runtime_1[\\"default\\"].TypeAliasDeclaration(\\"B\\", function (typeFactory) {
    var type = A;
    return type;
}, []);
"
`;
